src/linksys-addfwhdr.c

```diff
@@ -0,0 +1,191 @@
+/*
+	Linksys e8350 v1 firmware header generator
+*/
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <stdarg.h>
+#include <getopt.h>
+#include <time.h>
+#include <sys/time.h>
+#include "cyg_crc.h"
+#define AC2350          20
+
+#define CYBERTAN_VERSION        "v1.0.03"
+#define SERIAL_NUMBER   "003"
+#define MINOR_VERSION   ""
+#define BUILD_KEYWORD   " B"
+#define BUILD_NUMBER    SERIAL_NUMBER
+#define BETA_VERSION    " "
+#define CYBERTAN_UBOOT_VERSION      "v1.0"
+
+//add for AC2350 F/W header
+#define FWHDR_MAGIC_STR "CHDR"
+#define FWHDR_MAGIC 0X52444843
+struct cbt_fw_header
+{
+	unsigned int magic;             /* "CHDR" */
+	unsigned int len;               /* Length of file including header */
+	unsigned int crc32;             /* 32-bit CRC */
+	unsigned int res;
+};
+
+#define MAX_BUF	1024
+#define CRC32_INIT_VALUE 0xffffffff	/* Initial CRC32 checksum value */
+
+#ifndef TYPEDEF_UINT8
+typedef unsigned char   uint8;
+#endif
+
+#ifndef TYPEDEF_UINT16
+typedef unsigned short  uint16;
+#endif
+
+#ifndef TYPEDEF_UINT32
+typedef unsigned int    uint32;
+#endif
+
+typedef uint32_t cyg_uint32;
+typedef uint16_t cyg_uint16;
+
+int fd, fd_w;
+
+void die(const char * str, ...)
+{
+	va_list args;
+	va_start(args, str);
+	vfprintf(stderr, str, args);
+	fputc('\n', stderr);
+	exit(1);
+}
+
+int fill_null0(int size)
+{
+	unsigned char buf[1];
+	int i;
+
+	fprintf(stderr,"Fill null\n");
+
+	buf[0] = 0xff;
+	for (i=0 ; i< size; i++ )
+		if (write(fd_w, buf, 1) != 1)
+			return 0;
+
+	return 1;
+}
+
+long file_open(const char *name)
+{
+	struct stat sb;
+	if ((fd = open(name, O_RDONLY, 0)) < 0){
+		die("Unable to open `%s' : %m", name);
+	}
+
+	if (fstat (fd, &sb))
+		die("Unable to stat `%s' : %m", name);
+
+	return sb.st_size;
+}
+
+void usage(void)
+{
+	die("Usage: addfwhdr [-i|--input] sysupgrade.o [-o|--output] code.bin\n");
+}
+
+int main(int argc, char ** argv)
+{
+	unsigned int input_size,c;
+	char *input_file=NULL, *output_file=NULL;
+	int opt;
+	int option_index=0;
+	int garbage = 0;
+	char *buf = NULL;
+	extern char *optarg;
+	extern int optind, opterr, optopt;
+	
+	struct cbt_fw_header *fwhdr;
+	uint32 crc;	
+
+	static struct option long_options[] =
+	{
+		{"input", 1, 0, 'i'},
+		{"output", 1, 0, 'o'},
+		{"garbage", 0, 0, 'g'},
+		{0, 0, 0, 0}
+	};
+
+	printf("\n---------- add fw header --------\n");
+
+	fwhdr  = malloc(sizeof(struct cbt_fw_header));
+	memset(fwhdr, 0, sizeof(struct cbt_fw_header));	
+
+	while(1){
+		opt = getopt_long(argc, argv, "i:o:g",long_options, &option_index);
+		if(opt == -1)
+			break;
+		switch(opt){
+			case 'h' : 
+				usage(); break;
+			case 'i' :
+				input_file = optarg;
+				printf("input file is [%s]\n",input_file); break;
+			case 'o' :
+				output_file = optarg;
+				printf("output file is [%s]\n",output_file); break;
+			case 'g' :
+				garbage = 1; break;
```

Comment:
break on the next line. (same for prev entry)

Reply:
@Ansuel 
Done for all breaks. Would that be correct?